webapp/api_handlers.go

```diff
@@ -374,3 +377,122 @@ func handleAPIDiffPost(w http.ResponseWriter, r *http.Request) {
 	}
 	w.Write(bytes)
 }
+
+func apiManifestHandler(w http.ResponseWriter, r *http.Request) {
+	sha, err := ParseSHAParamFull(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ctx := appengine.NewContext(r)
+	if manifest, err := getManifestForSHA(ctx, sha); err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+	} else if manifest == nil {
+		http.NotFound(w, r)
+	} else {
+		w.Header().Add("content-type", "application/json")
+		w.Write(manifest)
+	}
+}
+
+func getManifestForSHA(ctx context.Context, sha string) (manifest []byte, err error) {
+	// Fetch models.Token entity for GitHub API Token.
+	tokenKey := datastore.NewKey(ctx, "Token", "github-api-token", 0, nil)
+	var token models.Token
+	datastore.Get(ctx, tokenKey, &token)
+
+	// Search for the PR associated with the SHA.
+	const githubSearch = `https://api.github.com/search/issues?q=SHA:%s+user:w3c+repo:web-platform-tests`
+	url := fmt.Sprintf(githubSearch, sha)
+	var body []byte
+	if body, err = fetchGitHubURL(ctx, url, token.Secret); err != nil {
+		return nil, err
+	}
+
+	var queryResults map[string]*json.RawMessage
+	if err = json.Unmarshal(body, &queryResults); err != nil {
+		return nil, err
+	}
+	var issues []map[string]*json.RawMessage
+	if err = json.Unmarshal(*queryResults["items"], &issues); err != nil {
+		return nil, err
+	}
+	if len(issues) < 1 {
```

Comment:
I'd like to assert len(issues)==1 and perhaps log all anomalies.

Reply:
Assertions should be for things that the programmer thinks is impossible, but logging which requests don't find a manifest SGTM. Except, do we have any server side logs at all currently? The less we log the less we need to think about PII and that kind of thing.

Reply:
AppEngine has built-in logging support. We can just use the standard logging facility provided by the language, and all the logs will go to Google Cloud Stackdriver which has a fancy UI for filtering.

Reply:
Regarding PII, AppEngine logs requests already. We won't and can't add extra PII if we log some error messages.